docs/pages/blog/turbo-1-10-0.mdx

```diff
@@ -0,0 +1,137 @@
+Previously, you would need to copy code yourself and manually add dependencies to your new workspace. With Generators,
+we've automated the process of creating new workspaces and integrating them into your repository.
+
+You can [create a blank workspace from scratch](/repo/docs/core-concepts/monorepos/code-generation#add-an-empty-workspace), [copy an existing workspace](/repo/docs/core-concepts/monorepos/code-generation#copy-an-existing-workspace), and even create custom generators
+for more control. Custom generators leverage [Plop](https://plopjs.com/) configurations, a common template definition strategy used in the JavaScript ecosystem.
+
+[Custom generators](/repo/docs/core-concepts/monorepos/code-generation#custom-generators) can also be used to scaffold more granular sets of code than workspaces. You can significantly speed up tasks like:
+
+- [Adding new components to your UI library](https://github.com/vercel/turbo/blob/main/examples/basic/packages/ui/turbo/generators/config.ts)
+- Creating new unit tests
+- Generating a new MDX blog post (In fact, this very post was created with a [Turborepo generator](https://github.com/vercel/turbo/blob/main/docs/turbo/generators/config.ts)!)
+
+Get started with your first generator by running:
+
+```
+turbo gen
+```
+
+For more, check out the [documentation](/repo/docs/core-concepts/monorepos/code-generation).
+
+## Improved Environment Variable Inputs
+
+### Support for `.env` files
+
+Developers and frameworks commonly use `.env` files to load environment variables into their development environments. This can cause problems when running with turbo, as these variables can be [invisible](/repo/docs/core-concepts/caching/environment-variable-inputs#invisible-environment-variables), and are not easily
+included in the hash for your task. This problem could cause an unintended cache hit while working locally.
+
+Turbo now makes this much easier with automatic support for `.env` files.
+
+To ensure that `turbo` includes these variables in your hash, use the `dotEnv` key. For example, here is a configuration for a Next.js application:
+
+```json
+{
+  "$schema": "https://turbo.build/schema.json",
+  "globalDotEnv": [".env"],
+  "pipeline": {
+    "build": {
+      "dotEnv": [".env.production.local", ".env.local", ".env.production", ".env"]
+    },
+    "dev": {
+      "dotEnv": [".env.development.local", ".env.local", ".env.development", ".env"]
+    },
+    "test": {
+      "dotEnv": [".env.test.local", ".env.test", ".env"]
+    }
+  }
+}
+```
+
+<Callout>
+  This feature does **not** handle system environment variables. If you are not using `.env` files outside of development, make sure your environment variables are still enumerated in either [env](/repo/docs/core-concepts/caching/environment-variable-inputs#pipelinetaskenv), or [globalEnv](/repo/docs/core-concepts/caching/environment-variable-inputs#globalenv) to ensure they are included in your hash.
```

Comment:
what are "system environment variables"?

Reply:
Yea good call - we definitely should. I'll add them to the new env var docs